BridgeSDK/BridgeAPI/SBBBridgeInfoProtocol.h

```diff
@@ -0,0 +1,81 @@
+//
+//  SBBBridgeInfoProtocol.h
+//  BridgeSDK
+//
+// Copyright (c) 2017, Sage Bionetworks. All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without modification,
+// are permitted provided that the following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice, this
+// list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice,
+// this list of conditions and the following disclaimer in the documentation and/or
+// other materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder(s) nor the names of any contributors
+// may be used to endorse or promote products derived from this software without
+// specific prior written permission. No license is granted to the trademarks of
+// the copyright holders even if such marks are included in this software.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@protocol SBBBridgeInfoProtocol <NSObject>
+
+/**
+ Study identifier used to setup the study with Bridge
+ */
+@property (nonatomic, readonly, copy) NSString *studyIdentifier;
+
+/**
+ Name of .pem certificate file to use for uploading to Bridge (without the .pem extension)
+ */
+@property (nonatomic, readonly, copy) NSString * _Nullable certificateName;
+
+/**
+ If using BridgeSDK's built-in caching, number of days ahead to cache
+ */
+@property (nonatomic, readonly) NSInteger cacheDaysAhead;
+
+/**
+ If using BridgeSDK's built-in caching, number of days behind to cache
+ */
+@property (nonatomic, readonly) NSInteger cacheDaysBehind;
+
+/**
+ Environment to load
+ */
+@property (nonatomic, readonly) SBBEnvironment environment;
+
+/**
+ Keychain service name.
+ */
+@property (nonatomic, readonly, copy) NSString * _Nullable keychainService;
```

Comment:
Well, my assumption was that they'd use something like this between themselves (via BridgeAppSDK in the app, and an extension to the SBBBridgeInfoProtocol in the keyboard that adds the same members). It just doesn't make sense for BridgeSDK itself to know about these because it wouldn't use them itself (only indirectly via the auth delegate shared between the app and the keyboard).